gcip/addons/container/jobs/crane.py

```diff
@@ -48,3 +48,53 @@ def copy(
         job.prepend_scripts(*docker_client_config.get_shell_command())
 
     return job
+
+
+def push(
+    dst: str,
+    image_path: Optional[str] = None,
+    image_name: Optional[str] = None,
+    *,
+    docker_client_config: Optional[DockerClientConfig] = None,
+    crane_image: Optional[Union[Image, str]] = None,
+) -> Job:
+    """
+    Creates a job to push container image to remote container registry with `crane`.
+
+    The image to copy must be in a `tarball` format. It gets validated with crane
+    and is pushed to `dst` destination registry.
+
+    Args:
+        dst (str): Registry URL to copy container image to.
+        image_path (Optional[str], optional): Path where to find the container image.
+            If `None` it defaults internally to `PredefinedVariables.CI_PROJECT_DIR`. Defaults to None.
+        image_name (Optional[str], optional): Container image name, searched for in `image_path` and gets `.tar` appended.
+            If `None` it defaults internally to `PredefinedVariables.CI_PROJECT_NAME`. Defaults to None.
+        docker_client_config (Optional[DockerClientConfig], optional): Creates the Docker configuration file base on objects settings,
+            used by crane to authenticate against given registries. Defaults to None.
+        crane_image (Optional[Union[Image, str]], optional): Container image which contains `crane` command.
+            Defaults to PredefindedImages.CRANE.
+
+    Returns:
+        Job: Returns a `gcip.Job`, with neccessary configuration to push a container image stored as a `tarball` to a remote registry.
+    """
+    if not crane_image:
```

Comment:
Please set the default values in the signature of this method.
* You will spare six lines of code without necessary logic.
* It is much more obvious for users of the method what the default values of those method arguments are.

There is no need to be anxious about those values are then class wide available. Those values are not meant to be modified at object level so there should be no side effect between objects of this class.

Reply:
Hell yeah, i did that. Thank you.